test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/DerivedTypes.cs

```diff
@@ -0,0 +1,360 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
```

Comment:
I'm not sure why we need all of these derived classes in this file, seems to be a sign of code smell...